psycopg/psycopg/_py_transformer.py

```diff
@@ -313,12 +313,11 @@ def load_rows(self, row0: int, row1: int, make_row: RowMaker[Row]) -> list[Row]:
 
         return records
 
-    def load_row(self, row: int, make_row: RowMaker[Row]) -> Row | None:
-        if not (res := self._pgresult):
-            return None
+    def load_row(self, row: int, make_row: RowMaker[Row]) -> Row:
+        assert (res := self._pgresult)
 
         if not 0 <= row < self._ntuples:
-            return None
+            raise IndexError
```

Comment:
A message could do here
```suggestion
            raise IndexError("record index out of range")
```

Reply:
I also forgot to change `transform.pyx`. I should unify error handling, I will do it later.

Reply:
Now errors are raised in the same way as in `Transformer.load_rows`